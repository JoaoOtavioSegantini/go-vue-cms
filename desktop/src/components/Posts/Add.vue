<template>
  <div class="container-fluid">
    <div class="row">
      <div class="col-1 nav">
        <p class="display-6">
          <small>AÇÕES</small>
        </p>

        <a href="/posts" class="btn btn-secondary" title="Voltar">
          <i class="fa fa-arrow-circle-left"></i>
        </a>
      </div>
      <div class="col-11">
        <div class="card">
          <div class="card-body">
            <h5 class="title">
              <i class="fa fa-file-text-o"></i> Novo artigo
              <small>Inclusão de novo artigo no blog</small>
            </h5>

            <form @submit.prevent="save()">
              <div class="form-group">
                <label for="title">Título</label>
                <input
                  type="text"
                  class="form-control"
                  id="title"
                  placeholder="Título do artigo"
                  v-model="post.title"
                />
              </div>
              <div class="form-group">
                <label for="slug">URL</label>
                <input
                  type="text"
                  class="form-control"
                  id="slug"
                  placeholder="Url do artigo"
                  v-model="post.slug"
                />
              </div>
              <div class="form-group">
                <label for="body">Conteúdo</label>
                <ckeditor :editor="editor" v-model="editorData"></ckeditor>
              </div>
              <button type="submit" class="btn btn-primary">Salvar</button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import ClassicEditor from "@ckeditor/ckeditor5-build-classic/build/ckeditor";
export default {
  data() {
    return {
      post: {},
      editor: ClassicEditor,
      editorData: "<p>Content of the editor.</p>",
    };
  },
  methods: {
    save() {
      this.post.body = this.editorData;
      // eslint-disable-next-line no-unused-vars
      this.$store.dispatch("createPost", this.post).then((_res) => {
        this.$router.push({ path: "/posts" });
      });
    },
  },
};
</script>

<style>
.ck-editor__editable {
  min-height: 260px;
}
</style>
