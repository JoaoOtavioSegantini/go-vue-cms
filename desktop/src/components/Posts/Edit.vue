<template>
  <div class="container-fluid">
    <div class="row">
      <div class="col-1 nav">
        <p class="display-6">
          <small>AÇÕES</small>
        </p>

        <a :href="'/posts/' + post.ID" class="btn btn-secondary" title="Voltar">
          <i class="fa fa-arrow-circle-left"></i>
        </a>
      </div>
      <div class="col-11">
        <div class="card">
          <div class="card-body">
            <h5 class="title">
              <i class="fa fa-file-text-o"></i> {{ post.title }}
              <small>Edição de artigo do blog</small>
            </h5>

            <form @submit.prevent="save()">
              <div class="form-group">
                <label for="title">Título</label>
                <input
                  type="text"
                  class="form-control"
                  id="title"
                  placeholder="Título da página"
                  v-model="post.title"
                />
              </div>
              <div class="form-group">
                <label for="slug">URL</label>
                <input
                  type="text"
                  class="form-control"
                  id="slug"
                  placeholder="Url da página"
                  v-model="post.slug"
                />
              </div>
              <div class="form-group">
                <label for="body">Conteúdo</label>
                <ckeditor :editor="editor" v-model="post.body"></ckeditor>
              </div>
              <button type="submit" class="btn btn-primary">Salvar</button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
const ClassicEditor = require("@ckeditor/ckeditor5-build-classic");
export default {
  data() {
    return {
      editor: ClassicEditor,
    };
  },
  computed: {
    post() {
      return { ...this.$store.state.Posts.onePost };
    },
  },
  methods: {
    save() {
      // eslint-disable-next-line no-unused-vars
      this.$store.dispatch("updatePost", this.post).then((_res) => {
        this.$router.push({ path: "/posts/" + this.post.ID });
      });
    },
  },
  mounted() {
    this.$store
      .dispatch("getPost", this.$route.params.id)
      // eslint-disable-next-line no-unused-vars
      .then((_res) => {
    //    console.log(this.post.body);
      })
      .catch((err) => {
        console.error(err.stack);
      });
  },
};
</script>

<style>
.ck-editor__editable {
  min-height: 260px;
}
</style>
