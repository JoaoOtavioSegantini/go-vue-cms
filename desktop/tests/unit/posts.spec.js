import Posts from "@/components/Posts.vue";
import { mount } from "@vue/test-utils";
import { createStore } from "vuex";

describe("Posts.vue", () => {
  const createVuexStore = () => {
    return createStore({
      state() {
        return {
          Posts: {
            all: [{ ID: "1", title: "Post 1", body: "A simple post body" }],
          },
        };
      },
      mutations: {
        updatePostsList(state) {
          state.Posts = [];
        },
      },
      actions: {
        listPosts(ctx) {
          const res = mockGet();
          ctx.commit("updatePostsList", res);
        },
      },
    });
  };

  function factory() {
    const store = createVuexStore();
    return mount(Posts, {
      global: {
        plugins: [store],
      },
    });
  }

  const mockGet = jest.fn();

  jest.mock("axios", () => ({
    get: () => mockGet(),
  }));

  it("renders props when passed", () => {
    const msg = "AÇÕES Artigos Gerenciamento de artigos do blog#titlebody";

    const wrapper = factory();

    expect(mockGet).toHaveBeenCalled();
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.find(".card").exists()).toBe(true)
    expect(wrapper.find(".card-body").exists()).toBe(true)
    expect(wrapper.find(".title").exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot();
  });
});
