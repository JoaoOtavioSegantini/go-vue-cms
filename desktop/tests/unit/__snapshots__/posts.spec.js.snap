// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts.vue renders props when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col-1 nav\\">
      <p class=\\"display-6\\"><small>AÇÕES</small></p><a href=\\"/posts/add\\" class=\\"btn btn-secondary\\" title=\\"Novo artigo\\"><i class=\\"fa fa-plus\\"></i></a>
    </div>
    <div class=\\"col\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> Artigos <small>Gerenciamento de artigos do blog</small></h5>
          <table class=\\"table\\" aria-label=\\"uma tabela contendo uma lista de posts\\">
            <thead>
              <tr>
                <th scope=\\"col\\">#</th>
                <th scope=\\"col\\">title</th>
                <th scope=\\"col\\">body</th>
                <th scope=\\"col\\"></th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th scope=\\"row\\">1</th>
                <td>Post 1</td>
                <td>
                  <div class=\\"cut-text\\">A simple post body</div>
                </td>
                <td class=\\"text-right\\"><a href=\\"/posts/1\\" class=\\"btn btn-primary btn-sm\\">ver</a></td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
